crates/hdk/src/lib.rs

```diff
@@ -1,20 +1,41 @@
-//! ## Holochain Development Kit (HDK)
+//! The Holochain Development Kit (HDK) provides high and low level functions for writing Holochain applications.
 //!
-//! The HDK exists to make working with WASM in holochain much easier.
+//! Functions of a Holochain application (hApp) can be organized into reusable components. In Holochain terminology these components are called "zomes".
+//! One or multiple zomes are compiled into a WebAssembly (WASM) binary, referred to as a DNA. All of the DNAs of an application are bundled to a hApp.
 //!
-//! Hopefully you don't notice the WASMness at all and it just feels like Rust 🦀
+//! hApps can be installed on a device that's running a so-called conductor, Holochain's runtime. Clients can then call each zome's functions via Remote Procedure Calls (RPC).
+//! Every zome function specifies the response it returns to the client as part of a zome's code.
+//! [More info on Holochain's architecture](https://developer.holochain.org/concepts/2_application_architecture/)
+//!
+//! hApps are required to be deterministic and shareable artifacts with a unique hash value, and must run in a sandboxed environment on the hosting system.
+//! Instead of writing and maintaining a custom format and specification for these artifacts as well as a runtime environment to execute them,
+//! Holochain makes use of WASM as the format of its applications. WASM binaries meet the aforementioned requirements as per the
+//! [WebAssembly specification](https://webassembly.github.io/spec/core/).
+//!
+//! Low-level communication between the conductor and WASM binaries, like typing and serialization of data, is encapsulated by the HDK.
+//! Using the HDK, hApp developers can focus on their application's logic. [Learn more about WASM in Holochain.](https://github.com/holochain/holochain/blob/develop/crates/hdk/ON-WASM.md)
+//!
+//! To start developing hApps, there's a [**hApp build tutorial**](https://github.com/holochain/happ-build-tutorial).
+//!
+//! # Example zomes 🍭
+//!
+//! The HDK is used in all the WASMs used to test Holochain itself.
+//! As they are used directly by tests in CI they are guaranteed to compile and work for at least the tests we define against them.
+//!
+//! At the time of writing there were about 40 example/test WASMs that can be browsed [on github](https://github.com/holochain/holochain/tree/develop/crates/test_utils/wasm/wasm_workspace).
+//!
```

Comment:
Note: It would also be nice to provide link to a place people can find other sample apps written in the ecosystem. But we don't have a good central spot for that yet. :(

Reply:
Yes, that would be extra cool. For the moment the test WASMs are the only official collection of examples that I know of. That's the biggest gap in our documentation AFAI can see, some advanced examples and topics.